mobile-save-for-later-user-deletion/src/main/scala/coml/gu/sfl/userdeletion/db/SflDynamoDb.scala

```diff
@@ -4,20 +4,24 @@ import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
 import com.amazonaws.services.dynamodbv2._
 import com.amazonaws.services.dynamodbv2.model.DeleteItemResult
 import org.scanamo.{Scanamo, Table}
-import org.scanamo.auto._
 import org.scanamo.syntax._
 import com.gu.sfl.Logging
 import com.gu.sfl.persistence.{DynamoSavedArticles, PersistenceConfig}
 import com.gu.sfl.userdeletion.model.UserDeleteMessage
+import org.scanamo.DeleteReturn.OldValue
+import org.scanamo.generic.auto.genericDerivedFormat
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
 
 class SflDynamoDb(persistanceConfig: PersistenceConfig) extends Logging {
 
   private val table = Table[DynamoSavedArticles](persistanceConfig.tableName)
-  private val client: AmazonDynamoDB = AmazonDynamoDBClient.builder().withCredentials(DefaultAWSCredentialsProviderChain.getInstance()).build()
+  private val client = DynamoDbClient.create()
   private val scanamo = Scanamo(client)
   def deleteSavedArticleasForUser(user: UserDeleteMessage) = {
     logger.info(s"Deleting record for user id: ${user.userId}")
-    logger.info(Option(scanamo.exec(table.delete("userId" -> user.userId))).fold(s"Unable to delete record for user ${user.userId}")((_: DeleteItemResult) => s"Deleted record for ${user.userId}") )
+    val dbResponse = scanamo.exec(table.deleteAndReturn(OldValue)("userId" === user.userId))
+      .fold(s"Unable to delete record for user ${user.userId}")((_) => s"Deleted record for ${user.userId}")
```

Comment:
Very small nits here, but:

```suggestion
    val dbResponse = scanamo.exec(table.deleteAndReturn(Nothing)("userId" === user.userId))
      .fold(s"Unable to delete record for user ${user.userId}")(_ => s"Deleted record for ${user.userId}")
```

...as you don't _use_ the old value, you can use `DeleteReturn.Nothing` rather than `DeleteReturn.OldValue`. You also don't need the extra parentheses around the underscore for the success-handling function's input!

I think some of the AWS SDk v1 imports at the top of the file can be cleared as well!

Reply:
Thanks updated in the next commit